src/main/java/org/myrobotlab/service/abstracts/AbstractSpeechSynthesis.java

```diff
@@ -1068,17 +1065,17 @@ public boolean isMute() {
 
   @Override
   public void setLocale(String code) {
-    locale = new Locale(code);
+    Runtime.getInstance().setLocale(code);
   }
 
   @Override
   public String getLanguage() {
-    return locale.getLanguage();
+    return Runtime.getInstance().getLanguage();

```

Comment:
This should return the language associated with the currently chosen voice.  not runtime locale

Reply:
it supports the LocaleProvider interface - a setting getting voice operates on different data

Reply:
Remove the interface then.  This is wrong.  Ifyou set the langauge of speech synthesis, it should choose a voice that supports the desired langauge.. or thow and exception if the speech synthesis doesn't support that language.  This is very confusing now.

Reply:
Get language should return the current language that the current voice is for.  This is returning the runtime langauge which is incorrect.